setup.py

```diff
@@ -9,6 +9,13 @@
 
 here = path.abspath(path.dirname(__file__))
 
+filename = path.join(here, 'kivy_garden', 'flower', '_version.py')
+# change this                              ^^^^^^
+locals = {}
+with open(filename, "rb") as fh:
+    exec(compile(fh.read(), filename, 'exec'), globals(), locals)
+__version__ = locals['__version__']
```

Comment:
I'm a little concerned with this part. I'm not sure this is needed, as opposed to simply importing the project version. That would not cause any imports to kivy.

Does `pip install xxx.zip` not work when you import the project in `setup.py`? I do that quite often and haven't had issues yet.

Reply:
If I understand it correctly the question concerns if one can write
```python
from kivy_garden.flower import __version__
```
or not. I will try. Here are the steps:
```
python -m virtualenv kivytest
source kivytest/bin/activate
pip install flower-master.zip
```
then I get the following error
```
ERROR: Command errored out with exit status 1:
     command: /home/gudrun/davidkl/Documents/kivytest/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-i7k3r042/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-i7k3r042/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-req-build-i7k3r042/pip-egg-info
         cwd: /tmp/pip-req-build-i7k3r042/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-i7k3r042/setup.py", line 8, in <module>
        from kivy_garden.flower import __version__
      File "/tmp/pip-req-build-i7k3r042/kivy_garden/flower/__init__.py", line 9, in <module>
        from kivy.uix.label import Label
    ModuleNotFoundError: No module named 'kivy'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```
Of course one can just install `kivy` first and everything is fine. But it is an issue if one want to distribute a package on PyPI and want users to able to install it with 
```
pip install my_package
```
preferably `kivy` and `kivy_garden`-flowers should be listed as dependencies.

Reply:
I also tried the following in `setup.py`
```python
from kivy_garden.flower._version import __version__
```
but with the same result.
```
Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-824vpawd/setup.py", line 7, in <module>
        from kivy_garden.flower._version import __version___
      File "/tmp/pip-req-build-824vpawd/kivy_garden/flower/__init__.py", line 9, in <module>
        from kivy.uix.label import Label
    ModuleNotFoundError: No module named 'kivy'
```

Reply:
I guess we can manually import just that file directly, using pythons importlib. But since this is similar to how cython flower does it, we can keep it consistent.